lib/improver/wxcode/wxcode_decision_tree.py

```diff
@@ -73,6 +73,29 @@ def wxcode_decision_tree():
     """
     queries = {
 
+        'lightning': {
+            'succeed': 'lightning_cloud',
+            'fail': 'heavy_precipitation',
+            'probability_thresholds': [0.5],
+            'threshold_condition': '>=',
+            'condition_combination': '',
+            'diagnostic_fields':
+                ['probability_of_number_of_lightning_flashes_'
+                 'per_unit_area_in_vicinity_above_threshold'],
+            'diagnostic_thresholds': [AuxCoord(0.0, units="m-2")],
+            'diagnostic_conditions': ['above']},
```

Comment:
Suggestion: 
I think what we've done here could be simplified somewhat (but I might be wrong). Rather than treating `lightning` as a special case, and worrying about how we'd cope with optional nodes when they're in the middle of the tree, how about we do something like:
```
'optional': True,
'missing_action': 'fail'
```
That way we could potentially have optional nodes anywhere, just so long as we redirect to a part of the tree which is hierarchically below.

Reply:
(I suspect this might be better to discuss on the phone, rather than GitHub)

Reply:
It OK I can see what you mean and it looks like a good idea. I will try this and see how it goes.

Reply:
OK, I've added diagnostic_missing_action to the decision tree queries. It could point to a different node but I've kept it as fail or succeed because
1. This is sufficient for what I want to do.
2. If it just pointed to any node there would be the possibility of getting into an infinite loop which I don't think is possible with the current set up.

I think this is now much better than before as we don't need to maintain a with or without lightning decision tree. The only thing I'm still not keen on is the start_node is hard coded in weather_symbols.py. I feel this should be something that is set in the weather decision tree and picked out by weather_symbols. What do you think?

Reply:
Sounds good to me - agreed with leaving as succeed/fail as well, also very happy that we don't need another tree. 👍 

I agree that having the start node encoded in `weather_symbols.py` is a bit messy. I'd probably just suggest adding a constant to the `wxcode_decision_tree` and `wxcode_decision_tree_global` modules, probably `ROOT_NODE` or similar. Does that sound sensible?

Reply:
Changed to START_NODE to match the name used in weather_symbols